src/lib/expression/evaluation/like_matcher.hpp

```diff
@@ -82,19 +82,23 @@ class LikeMatcher {
    *    LikeMatcher{"%hello%"}.resolve(false, [](const auto& matcher) {
    *        std::cout << matcher("He said hello!") << std::endl;
    *    }
+   *
+   * The first template type T defaults to pmr_string, but can also be std::string_view (e.g., for fixed string
+   * dictionary segments). The template type for the functor is the second template argument, since it usually
+   * deduced while T is not decuded automatically.
    */
-  template <typename Functor>
+  template <typename T = pmr_string, typename Functor>
   void resolve(const bool invert_results, const Functor& functor) const {
     if (std::holds_alternative<StartsWithPattern>(_pattern_variant)) {
       const auto& prefix = std::get<StartsWithPattern>(_pattern_variant).string;
-      functor([&](const pmr_string& string) -> bool {
+      functor([&](const T& string) -> bool {
```

Comment:
Can't you accept auto here and get rid of `T`?

Reply:
Wow. Did not know that's possible. 

Is that guaranteed to work as expected? I wonder if `resolve`'s inner functions are instantiated for several every type now instead of `resolve` being instantiated twice for `pmr_string` and `std::string_view`?

Reply:
See below. Auto is now used.